pkgs/development/libraries/qt-5/5.7/qtbase/default.nix

```diff
@@ -135,61 +109,118 @@ stdenv.mkDerivation {
+    ${lib.optionalString stdenv.isDarwin "-no-use-gold-linker"}
+    -${lib.optionalString stdenv.isDarwin "no-"}xcb
+    -no-eglfs
+    -no-kms
+    -no-gbm
+    -no-directfb
+    -no-linuxfb
+    -no-mirclient
+    -no-openvg
+    ${lib.optionalString stdenv.isLinux "-qpa xcb"}
+    ${lib.optionalString stdenv.isDarwin "-qpa cocoa"}
+    -opengl desktop
+    -${lib.optionalString stdenv.isDarwin "no-"}libinput
+    -system-proxies
+
+    ${lib.optionalString stdenv.isDarwin "-no-framework"}
+    ${lib.optionalString stdenv.isDarwin "-securetransport"}
+
+    ${lib.optionalString stdenv.isDarwin "-no-libudev"}
+
+    ${lib.optionalString stdenv.isDarwin "-I${libtiff.dev}/include/ -L${libtiff.out}/lib"}
+    ${lib.optionalString stdenv.isDarwin "-I${libmng.dev}/include/ -L${libmng.out}/lib"}
+    ${lib.optionalString stdenv.isDarwin "-I${libwebp.out}/include/ -L${libwebp.out}/lib"}
+    ${lib.optionalString stdenv.isDarwin "-I${jasper.dev}/include/ -L${jasper.out}/lib"}
   '';
 
   # PostgreSQL autodetection fails sporadically because Qt omits the "-lpq" flag
   # if dependency paths contain the string "pq", which can occur in the hash.
   # To prevent these failures, we need to override PostgreSQL detection.
   PSQL_LIBS = lib.optionalString (postgresql != null) "-L${postgresql.lib}/lib -lpq";
 
-  propagatedBuildInputs = [
-    dbus glib libxml2 libxslt openssl pcre16 sqlite udev zlib
-
-    # Image formats
-    libjpeg libpng libtiff
-
-    # Text rendering
-    fontconfig freetype harfbuzz icu
-
-    # X11 libs
-    libX11 libXcomposite libXext libXi libXrender libxcb libxkbcommon xcbutil
-    xcbutilimage xcbutilkeysyms xcbutilrenderutil xcbutilwm
+  propagatedBuildInputs =
+  [
+    dbus libxml2 libxslt zlib pcre16 sqlite harfbuzz icu double_conversion
+    libjpeg libpng libtiff libmng libwebp jasper
   ]
-  ++ lib.optional mesaSupported mesa;
+  ++ lib.optional stdenv.isLinux [
+    openssl udev cups fontconfig freetype glib
+    libX11 libXcomposite libXext libXi libXrender libxkbcommon
+    libxcb xcbutil xcbutilimage xcbutilkeysyms xcbutilrenderutil xcbutilwm
+  ]
+  ++ lib.optional (null != mesa) mesa
+  ++ lib.optional stdenv.isDarwin
+  [
+    Cocoa CoreServices CoreGraphics  CoreText OpenGL AppKit
+    Security SystemConfiguration CFNetwork DiskArbitration IOKit
+  ]
+  ++ lib.optional (postgresql != null) postgresql;
 
   buildInputs =
-    [ gtk3 libinput ]
-    ++ lib.optional developerBuild gdb
-    ++ lib.optional (cups != null) cups
-    ++ lib.optional (mysql != null) mysql.lib
-    ++ lib.optional (postgresql != null) postgresql;
+  lib.optional stdenv.isLinux [ gtk3 libinput ]
+  ++ lib.optional developerBuild gdb
+  ++ lib.optional (null != mysql) mysql.lib
+  ++ lib.optional (null != postgresql) postgresql;
 
   nativeBuildInputs =
-    [ bison flex gperf lndir patchelf perl pkgconfig python ruby ];
+  [ bison flex gperf lndir patchelf perl pkgconfig python ruby bash ];
```

Comment:
@adolfogc `bash` is already included in `stdenv` so this should not be necessary.